File: redsheet-portal/src/app/database-models/rapport-plan.ts
import { required, maxLength, range } from '@rx/annotations';

export class RapportPlan {
    constructor(rapportPlan?: RapportPlan )  {
        let properties = [ "cultureCountryId", "rapportPlanId", "rapportPlanValue",];
        for (let property of properties)
            if (rapportPlan && rapportPlan[property])
                this[property] = rapportPlan[property];
    }
 
    @range(1,2147483647)
	cultureCountryId : number =   undefined;
 
	rapportPlanId : number =   0 ;
 
    @required()
    @maxLength(500)
	rapportPlanValue : string =   undefined;
}

File: RedSheet.DbEntities/Models/RapportPlan.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("RapportPlans", Schema = "dbo")]
	[Serializable]
	public partial class RapportPlan  
	{
 
	[Range(1, int.MaxValue)]
	public int CultureCountryId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int RapportPlanId { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string RapportPlanValue { get; set; } 

	public RapportPlan()
	{
    }

    }
}

